### ISSUE TITLE
Removing `outline_recursive` does not remove outline

### ISSUE THREAD

**Issue by marceline-cramer** [AmbientRun/Ambient#529]
Flowerpot adds `outline_recursive` onto selected entities and removes the `outline_recursive` component from just-unselected entities. Only one entity should have an outline at a time, and until a recent update that I haven't tracked down, that worked correctly. Here's what it looks like now:
![20230622_14h35m04s_grim](https://github.com/AmbientRun/Ambient/assets/56894066/5c97b208-1e50-42f7-a488-17ee58dd0744)

**Comment by FredrikNoren** [comment]
@marceline-cramer Yeah sorry it's a bug from a change we did to get scroll area to work. Pushed a fix, re-open if you still have problems though.

**Comment by marceline-cramer** [comment]
I'm still having the same issue. :(

The unit test you added passes on my machine, but the tiles stay highlighted. Curiously enough, I've noticed that ONLY tiles stay highlighted, not crops that occupy tiles. The difference that immediately comes to my mind between the two is that crop model entities are spawned entirely by the client, whereas tile entities are spawned by the server, even if the client adds its own rendering components to it.

Here's my code that updates the outlines: https://github.com/marceline-cramer/flowerpot-prototype/blob/main/src/client/player.rs#L236

I'm using commit 9e753d3 for both my runtime version and my ambient_api dependency revision, which is the latest commit at the time of writing.

**Comment by philpax** [comment]
I've also encountered this - I think the issue arises from this despawn query here

https://github.com/AmbientRun/Ambient/blob/main/crates/ecs/src/helpers.rs#L109-L110

which checks for *both* the recursive component and the children being removed, not just the former. 

This means that for our use cases, where we're just removing the outline from an existing entity and not despawning the entire entity, the outline persists.

I think this should be an easy enough fix if I've diagnosed the issue correctly.
